src/Sylius/Bundle/PaymentBundle/Form/Type/PaymentMethodChoiceType.php

```diff
@@ -22,35 +29,76 @@
  * @author Paweł Jędrzejewski <pawel@sylius.org>
  * @author Arnaud Langlade <arn0d.dev@gmail.com>
  * @author Mateusz Zalewski <mateusz.zalewski@lakion.com>
+ * @author Anna Walasek <anna.walasek@lakion.com>
  */
-class PaymentMethodChoiceType extends ResourceChoiceType
+class PaymentMethodChoiceType extends AbstractType
 {
     /**
+     * @var CompositeMethodsResolverInterface
+     */
+    protected $compositePaymentMethodsResolver;
+
+
+    /**
+     * @var RepositoryInterface
+     */
+    protected $repository;
+
+    /**
+     * @param CompositeMethodsResolverInterface $compositeShippingMethodsResolver
+     * @param RepositoryInterface $repository
+     */
+    public function __construct(
+        CompositeMethodsResolverInterface $compositeShippingMethodsResolver,
+        RepositoryInterface $repository
+    ) {
+        $this->compositeShippingMethodsResolver = $compositeShippingMethodsResolver;
+        $this->repository = $repository;
+    }
+
+    /**
      * {@inheritdoc}
      */
-    public function configureOptions(OptionsResolver $resolver)
+    public function buildForm(FormBuilderInterface $builder, array $options)
     {
-        parent::configureOptions($resolver);
-
-        $queryBuilder = function (Options $options) {
-            $repositoryOptions = [
-                'disabled' => $options['disabled'],
-            ];
+        if ($options['multiple']) {
+            $builder->addModelTransformer(new CollectionToArrayTransformer());
+        }
+    }
 
-            return function (PaymentMethodRepositoryInterface $repository) use ($repositoryOptions) {
-                return $repository->getQueryBuilderForChoiceType($repositoryOptions);
-            };
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        $choiceList = function (Options $options) {
+            $resolvedMethods = $this->repository->findAll();
```

Comment:
We should unfortunately use ``if...else`` statement, to avoid getting all payment methods every time.